helix-core/src/main/java/org/apache/helix/manager/zk/ParticipantManager.java

```diff
@@ -418,7 +418,16 @@ private void carryOverPreviousCurrentState() {
       String path = _keyBuilder.currentStates(_instanceName, session).getPath();
       LOG.info("Removing current states from previous sessions. path: " + path);
       _zkclient.deleteRecursively(path);
-      path = _keyBuilder.taskCurrentStates(_instanceName, session).getPath();
+    }
+
+    // Remove all previous task current state sessions
```

Comment:
I suggest you move the task CS sessions removal logic to a separate method. Just call it carryOverPreviousTaskCurrentState or cleanPreviousCurrentState.

We should avoid huge methods whenever possible.